python/python/Ice/Util.py

```diff
@@ -37,53 +36,28 @@ def initialize(
     initData: InitializationData | None = None,
 ) -> Communicator:
     """
-    Creates a new communicator.
+    Creates a new communicator. This function is provided for backwards compatibility. New code should call the
+    :class:`Communicator` constructor directly.
 
     Parameters
     ----------
     args : list of str, optional
-        The command-line arguments. This function parses arguments starting with ``--`` and one of the
-        reserved prefixes (Ice, IceSSL etc.) as properties for the new communicator. If there is an argument starting
-        with ``--Ice.Config``, this function loads the specified configuration file. When the same property is set in a
-        configuration file and through a command-line argument, the command-line setting takes precedence.
+        The command-line arguments.
     eventLoop : asyncio.AbstractEventLoop, optional
-        An asyncio event loop used to run coroutines and wrap futures. If provided, a new event loop adapter is created
-        and configured with the communicator. This adapter is responsible for executing coroutines returned by Ice
-        asynchronous dispatch methods and for wrapping Ice futures (from Ice Async APIs) into asyncio futures.
-        This argument and the `initData` argument are mutually exclusive. If the `initData` argument is provided, the
-        event loop adapter can be set using the :attr:`InitializationData.eventLoopAdapter` attribute.
+        An asyncio event loop used to run coroutines and wrap futures. This argument and the `initData` argument are
+        mutually exclusive.
     initData : InitializationData, optional
         Options for the new communicator. This argument and the `args` argument are mutually exclusive.
 
     Returns
     -------
     Communicator
         The new communicator.
-
-    Examples
-    --------
-
-    .. code-block:: python
-
-        with Ice.initialize(sys.argv, eventLoop=asyncio.get_running_loop()) as communicator:
-            greeter = VisitorCenter.GreeterPrx(communicator, "greeter:tcp -h localhost -p 4061")
-            await greeter.greetAsync()
     """
-
-    eventLoopAdapter = initData.eventLoopAdapter if initData else None
-
-    eventLoopAdapter = None
-    if initData:
-        eventLoopAdapter = initData.eventLoopAdapter
-    elif eventLoop:
-        eventLoopAdapter = AsyncIOEventLoopAdapter(eventLoop)
-
-    if args:
-        initData = InitializationData(properties=createProperties(args))
-
-    # initData can be None here, which is acceptable.
-    communicator = IcePy.Communicator(initData)
-    return Communicator(communicator, eventLoopAdapter)
+    if args is not None or eventLoop is not None:
+        return Communicator(args, eventLoop)
+    else:
+        return Communicator(initData=initData)
```

Comment:
maybe:

`return Communicator(initData=initData) if initData is not None else Communicator(args, eventLoop)`